compiler-cli/src/main.rs

```diff
@@ -526,6 +526,13 @@ fn initialise_logger() {
 }
 
 fn project_paths_at_current_directory() -> ProjectPaths {
+    let current_dir = get_current_directory().expect("Failed to get current directory");
+    let dir = get_project_root(current_dir).expect("Failed to get any gleam.toml file");
```

Comment:
which function are you meaning specifically, sorry? The `get_current_directory()` one? could probably mostly change that to not panic and just pass the `result` to `get_project_root` or refactor it out or something something, it does not appear to be used else where much.

or if you're meaning the whole function itself, it could still fail if you try run gleam say in `/etc` or whatever, it's not likely to find a gleam.toml in `etc` or `/`? May be able to pass the result back to the caller function though not yet certain where all the error handling is being dealt with yet. Happy to dig through that if that's wise though, wasn't wanting to change type signatures too much yet as I'm not sure all the implications ^^;

or if you were meaning something else, apologies, I may be being a bit dense hah. lemme know what you're thinking :).

Reply:
I meant `project_paths_at_current_directory`, which seems to not do what the name claims to do now.

We never panic on invalid user input so an error will need to be returned and printed in the case that the project root cannot be found.

Reply:
cool yeah, no worries! I'll refactor the function name the function and update the return sig to `Result<ProjectPaths>` and update code paths as appropriate if that sounds good ^^?

Reply:
Brilliant, thank you!

Reply:
hopefully my choice of name is suitable hah, any more thoughts/changes just let me know :)!

spotted some functions like the `list()` one in `compiler-cli/src/dependencies.rs` that i updated before that could probably be refactor to use the above function.

same maybe with `root_config()` in `compiler-cli/src/config.rs`. they're both kinda doing the same "build a ProjectPaths type from where ever gleam.toml is" calls so might make sense to refactor this as well, though may be other reasons for this I've not yet understood, let me know if that's a good idea or not.